src/thread_pool.rs

```diff
@@ -365,8 +387,8 @@ impl ThreadPool {
     pub fn inject(&'static self, job_ref: JobRef) {
         self.queue.push(job_ref);
 
-        // If the pool is inactive, we have to wake a thread to work on this.
-        if self.active_tally.load(Ordering::Relaxed) == 0 {
+        // If the was is inactive, we have to wake a thread to work on this.
+        if self.active_tally.load(Ordering::Relaxed) <= 1 {
```

Comment:
I still don't understand why we don't *always* wake a thread.  If there is one thread happily working on a long task, and we inject a second task, don't we want to wake a second thread?  Otherwise the task will sit in the queue until the first thread finishes the first task.

Reply:
Hmm, true. I'm not sure about the ideal semantics for this. I think you are right, this should always wake. At one point it did, not sure why I changed it.

Once we have correctness tests and benchmarks I will start to take more care with these sorts of things, i think.